src/components/controls/Dropdown.tsx

```diff
@@ -30,28 +31,22 @@ const usesCodeList = (
     );
 };
 
-class Dropdown extends React.Component
-<IDropdownProps | IDropdownWithCodeListProps> {
+class Dropdown extends React.Component<IDropdownProps | IDropdownWithCodeListProps> {
     onChange = (event: any) => {
         this.props.onChange(event.target.value);
     }
 
     public render() {
-        let dropDownItemList: IDropdownItem[];
+        let dropDownItemList: IDropdownItem[] =
+            this.props.allowEmpty ? [{ key: '', value: '-' }] : [];
```

Comment:
Is this the correct way to implement? How do you know what empty value is? What if empty value would be '' or null or undefined etc and not '-'?

What if you give emptyValue?: any: as props, if that is given, that will be the empty value and it will be used.

Reply:
Also that empty value should be used here I think:
{Boolean(selectedItem) ? selectedItem!.value : ''}

Reply:
Good comment, this was refactored now